packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -45,72 +63,235 @@ import {
+          userId: selfUser?.id,
+          locationId: (item as LocationType).id
         })
 
-        if (context.params.query.$sort[sort] === 1) {
-          data.reverse()
-        }
+        await trx.from<LocationAuthorizedUserType>(locationAuthorizedUserPath).insert({
+          ...(item as LocationType).locationAdmin,
+          userId: selfUser?.id,
+          locationId: (item as LocationType).id
+        })
       }
+
+      await trx.commit()
+
+      const location = await context.app.service(locationPath).get((item as LocationType).id)
+
+      result.push(location)
+    } catch (err) {
+      logger.error(err)
+      await trx.rollback()
+      if (err.code === 'ER_DUP_ENTRY') {
+        throw new Error('Name is in use.')
+      }
+      throw err
+    }
+  }
+
+  context.result = result.length === 1 ? result[0] : result
+}
+
+const updateLocation = async (context: HookContext<LocationService>) => {
+  if (!context.data || context.method !== 'patch') {
+    throw new BadRequest(`${context.path} service only works for data in ${context.method}`)
+  }
+
+  const data: LocationPatch = context.data as LocationPatch
+  const trx = await (context.app.get('knexClient') as Knex).transaction()
+
+  try {
+    const selfUser = context.params?.user
+
+    const oldLocation = await context.app.service(locationPath).get(context.id!)
+
+    if (!oldLocation.isLobby && data.isLobby) {
+      await context.service.makeLobby(trx, selfUser)
+    }
+
+    if (data.name) {
+      data.slugifiedName = slugify(data.name, { lower: true })
+    }
+
+    const updateData = JSON.parse(JSON.stringify(data))
+    delete updateData.locationSetting
+
+    await trx
+      .from<LocationDatabaseType>(locationPath)
+      .update(updateData)
+      .where({ id: context.id?.toString() })
+
+    if (data.locationSetting) {
+      await trx
+        .from<LocationSettingType>(locationSettingPath)
+        .update({
+          videoEnabled: data.locationSetting.videoEnabled,
+          audioEnabled: data.locationSetting.audioEnabled,
+          faceStreamingEnabled: data.locationSetting.faceStreamingEnabled,
+          screenSharingEnabled: data.locationSetting.screenSharingEnabled,
+          locationType: data.locationSetting.locationType || 'private'
+        })
+        .where({ id: oldLocation.locationSetting.id })
+    }
+
+    await trx.commit()
+
+    const location = await context.app.service(locationPath).get(context.id!)
+
+    context.result = location
+  } catch (err) {
+    logger.error(err)
+    await trx.rollback()
+    if (err.errors && err.errors[0].message === 'slugifiedName must be unique') {
+      throw new Error('That name is already in use')
     }
+    throw err
+  }
+}
+
+const checkIsLobby = async (context: HookContext<LocationService>) => {
+  if (context.id) {
+    const location = await context.app.service(locationPath).get(context.id)
+
+    if (location && location.isLobby) {
+      throw new Error("Lobby can't be deleted")
```

Comment:
Instead of Error it can be BadRequest. Same goes for other Error instances in this file, they can be replaced with appropriate feathers error https://feathersjs.com/api/errors.html#feathers-errors